src/operations/References.cpp

```diff
@@ -510,14 +351,37 @@ lsp::ReferenceResult WorkspaceFolder::references(const lsp::ReferenceParams& par
     {
         if (auto prefix = reference->prefix)
         {
-            if (auto importedModuleName = module->getModuleScope()->importedModules.find(prefix.value().value);
-                importedModuleName != module->getModuleScope()->importedModules.end())
+            auto requireInfo = findClosestAncestorModuleImport(*sourceModule, reference->prefix.value(), reference->prefixLocation->begin);
```

Comment:
I know I tried this at some point, but I don't remember why I ended up not doing it this way. It could've been because `Luau::findAstAncestryOfPosition` lets a visitor do the job under the hood and using a custom visitor will be more efficient. There may be a better reason for this decision, so I'll investigate when I get the chance.

Reply:
I think the efficiency gains are negligible with such a custom visitor vs. what's already implemented for `findAstAncestryOfPosition`. The main reason I prefer the Luau function is it will remain up-to-date if there are any issues with it, whilst we have to maintain our own visitor. However if it doesn't end up being simpler code then we can leave it.